frontend/utils/giteaApi.js

```diff
@@ -0,0 +1,229 @@
+ */
+export const migrateRepo = async (remoteRepo, uid, csrf) => {
+  const clone_addr = remoteRepo
+  const repo_name = urlToName(clone_addr)
+  const endpoint = `${giteaApiUrl}/repos/migrate?_csrf=${csrf}`
+  const giteaOptions = {
+    clone_addr,
+    uid,
+    repo_name,
+    mirror: false,
+    wiki: false,
+    private: false,
+    pull_requests: false,
+    releases: true,
+  }
+
+  const res = await fetch(endpoint, {
+    method: 'POST',
+    mode,
+    credentials: 'include',
+    headers,
+    body: JSON.stringify(giteaOptions),
+  })
+
+  return res.ok
+}
+/**
+ * delete the corresponding gitea repo for a project.
+ * @param repo {string}
+ * @param csrf {string}
+ * @returns {Promise<boolean>}
+ */
+export const deleteRepo = async (repo, csrf) => {
+  const endpoint = `${giteaApiUrl}/repos/${repo}?_csrf=${csrf}`
+  const res = await fetch(endpoint, {
+    method: 'DELETE',
+    mode,
+    credentials,
+    headers,
+  })
+
+  return res.ok
+}
+
+/**
+ * Get list of files in a gitea repo.
+ * @param repo {string}
+ * @param csrf {string}
+ * @returns {Promise<Array|null>}
+ */
+export const getRepoFiles = async (repo, csrf) => {
+  const endpoint = `${giteaApiUrl}/repos/${repo}/contents?ref=master&_csrf=${csrf}`
+  const res = await fetch(endpoint, {
+    method: 'GET',
+    mode,
+    credentials,
+    headers,
+  })
+
+  const body = await res.json()
+
+  if (body.hasOwnProperty('owner')) {
+    return []
+  } else {
+    return res.ok ? body : []
+  }
+}
+
+/**
+ * Get repo details
+ * @param fullname
+ * @returns {Promise<Object|null>}
+ */
+export const getRepo = async fullname => {
+  const endpoint = `${giteaApiUrl}/repos/${fullname}`
+
+  const res = await fetch(endpoint, {
+    method: 'GET',
+    credentials,
+    mode,
+    headers,
+  })
+  return res.ok ? await res.json() : null
+}
+
+/**
+ * Get the repos own by the current authenticated user
+ * @param csrf{string}
+ * @returns {Promise<any|*[]>}
+ */
+export const getUserRepos = async csrf => {
+  const endpoint = `${giteaApiUrl}/user/repos?_csrf=${csrf}`
+
+  const res = await fetch(endpoint, {
+    method: 'GET',
+    credentials,
+    mode,
+    headers,
+  })
+
+  return res.ok ? await res.json() : []
+}
+
+/**
+ * uploads a file to an existing gitea repo
+ * @param repo {string} full repo name, i.e., {user}/{repoName}
+ * @param path
+ * @param content: must be Base64 encoded
+ * @param csrf
+ * @returns {Promise<boolean>}
+ */
+export const uploadFile = async (repo, path, content, csrf) => {
+  const user = window.session.user
+  const endpoint = `${giteaApiUrl}/repos/${repo}/contents/${path}?_csrf=${csrf}`
+
+  const reqBody = {
+    author: {
+      email: user.email,
+      name: user.login,
+    },
+    branch: 'master',
+    committer: {
+      email: 'admins@kitspace.org',
+      name: 'Kitspace',
+    },
+    // content must be Base64 encoded
+    content: btoa(content),
+    message: `Automated commit on behalf of ${user.login} (${user.email})`,
```

Comment:
I think we can do better, that message is too long. I would suggest just `Upload {path}` when uploading a single file and `Upload multiple files` when we have that.

Reply:
Maybe even just `Add <filename>`, that's what Gitea seems to do.